File: 프로그래머스/2/42587. 프로세스/프로세스.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var queue: [(Int, Int)] = []
    
    for (i, j) in priorities.enumerated() {
        queue.append((i, j))
    }
    
    
    var count = 0
    
    while !queue.isEmpty {
        let current = queue.removeFirst()
        
        if queue.contains(where: { $0.1 > current.1} ) {
            queue.append(current)
        } else {
            count += 1
            
            if current.0 == location {
                return count
            } 
        }
    }
   
    
    return count
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    
    queue = []
    
    for i in range(len(priorities)):
        queue.append((priorities[i], i))
    
    while queue:
        higher = False
        current = queue.pop(0)
        
        for process in queue:
            if process[0] > current[0]:
                higher = True
                break
                
        if higher == True:
            queue.append(current)
        else:
            answer += 1
            if current[1] == location:
                return answer
           